panic on bad command input

```
$ buildifier foo ''
```

```
panic: runtime error: index out of range [0] with length 0

goroutine 1 [running]:
github.com/bazelbuild/buildtools/edit.parseCommands(0x4, {0xc0000101c0, 0x2, 0xc00011a460})
	edit/buildozer.go:821 +0x38f
github.com/bazelbuild/buildtools/edit.appendCommands(0xc00010c640, 0xc0000796c0, {0xc0000101c0, 0xc000169798, 0xc000115bd0})
	edit/buildozer.go:1096 +0x4f
github.com/bazelbuild/buildtools/edit.Buildozer(0xc00010c640, {0xc0000101c0, 0x2, 0x2})
	edit/buildozer.go:1208 +0x1a5
main.main()
	buildozer/main.go:122 +0x513
```

--- a/edit/buildozer.go
+++ b/edit/buildozer.go
@@ -815,9 +815,13 @@ func SplitOnSpaces(input string) []string {
 //   whitespace
 // - a target all commands that are parsed during one call to parseCommands
 //   should be applied on
-func parseCommands(opts *Options, args []string) (commands []command, targets []string) {
+func parseCommands(opts *Options, args []string) (commands []command, targets []string, err error) {
 	for _, arg := range args {
 		commandTokens := SplitOnSpaces(arg)
+		if len(commandTokens) == 0 {
+			return nil, nil, fmt.Errorf("empty command list")
+		}
+
 		cmd, found := AllCommands[commandTokens[0]]
 		if found {
 			checkCommandUsage(opts, commandTokens[0], cmd, len(commandTokens)-1)
@@ -1092,8 +1096,11 @@ func findBuildFiles(rootDir string) []string {
 
 // appendCommands adds the given commands to be applied to each of the given targets
 // via the commandMap.
-func appendCommands(opts *Options, commandMap map[string][]commandsForTarget, args []string) {
-	commands, targets := parseCommands(opts, args)
+func appendCommands(opts *Options, commandMap map[string][]commandsForTarget, args []string) error {
+	commands, targets, err := parseCommands(opts, args)
+	if err != nil {
+		return err
+	}
 	for _, target := range targets {
 		for _, buildFileName := range BuildFileNames {
 			if strings.HasSuffix(target, filepath.FromSlash("/"+buildFileName)) {
@@ -1113,9 +1120,10 @@ func appendCommands(opts *Options, commandMap map[string][]commandsForTarget, ar
 			commandMap[file] = append(commandMap[file], commandsForTarget{target, commands})
 		}
 	}
+	return nil
 }
 
-func appendCommandsFromFiles(opts *Options, commandsByFile map[string][]commandsForTarget, labels []string) {
+func appendCommandsFromFiles(opts *Options, commandsByFile map[string][]commandsForTarget, labels []string) error {
 	for _, fileName := range opts.CommandsFiles {
 		var reader io.Reader
 		if fileName == stdinPackageName {
@@ -1125,11 +1133,14 @@ func appendCommandsFromFiles(opts *Options, commandsByFile map[string][]commands
 			reader = rc
 			defer rc.Close()
 		}
-		appendCommandsFromReader(opts, reader, commandsByFile, labels)
+		if err := appendCommandsFromReader(opts, reader, commandsByFile, labels); err != nil {
+			return err
+		}
 	}
+	return nil
 }
 
-func appendCommandsFromReader(opts *Options, reader io.Reader, commandsByFile map[string][]commandsForTarget, labels []string) {
+func appendCommandsFromReader(opts *Options, reader io.Reader, commandsByFile map[string][]commandsForTarget, labels []string) error {
 	r := bufio.NewReader(reader)
 	atEOF := false
 	for !atEOF {
@@ -1139,8 +1150,7 @@ func appendCommandsFromReader(opts *Options, reader io.Reader, commandsByFile ma
 			err = nil
 		}
 		if err != nil {
-			fmt.Fprintf(opts.ErrWriter, "Error while reading commands file: %v", err)
-			return
+			return fmt.Errorf("error while reading commands file: %v", err)
 		}
 		line = strings.TrimSpace(line)
 		if line == "" || line[0] == '#' {
@@ -1149,11 +1159,16 @@ func appendCommandsFromReader(opts *Options, reader io.Reader, commandsByFile ma
 		args := strings.Split(line, "|")
 		if len(args) > 1 && args[1] == "*" {
 			cmd := append([]string{args[0]}, labels...)
-			appendCommands(opts, commandsByFile, cmd)
+			if err := appendCommands(opts, commandsByFile, cmd); err != nil {
+				return err
+			}
 		} else {
-			appendCommands(opts, commandsByFile, args)
+			if err := appendCommands(opts, commandsByFile, args); err != nil {
+				return err
+			}
 		}
 	}
+	return nil
 }
 
 func printRecord(writer io.Writer, record *apipb.Output_Record) {
@@ -1200,12 +1215,18 @@ func Buildozer(opts *Options, args []string) int {
 	}
 	commandsByFile := make(map[string][]commandsForTarget)
 	if len(opts.CommandsFiles) > 0 {
-		appendCommandsFromFiles(opts, commandsByFile, args)
+		if err := appendCommandsFromFiles(opts, commandsByFile, args); err != nil {
+			fmt.Fprintf(opts.ErrWriter, "error: %s\n", err)
+			return 1
+		}
 	} else {
 		if len(args) == 0 {
 			Usage()
 		}
-		appendCommands(opts, commandsByFile, args)
+		if err := appendCommands(opts, commandsByFile, args); err != nil {
+			fmt.Fprintf(opts.ErrWriter, "error: %s\n", err)
+			return 1
+		}
 	}
 
 	numFiles := len(commandsByFile)

